packages/radio-player/src/search-handler/search-index.ts

```diff
@@ -0,0 +1,139 @@
+import { TranscriptConfig, TranscriptEntryConfig } from '@internetarchive/transcript-view';
+import { Range, TranscriptEntryRange } from './search-models';
+
+class SearchIndexCache {
+  /**
+   * This gets populated as part of the search index build. It merges all of the transcript
+   * entries together so we can search it as a single document instead of a bunch of
+   * individual entries. This allows searches to cross over transcript entries.
+   *
+   * NOTE: When the mergedTranscript gets created, spaces are put between each transcript
+   * entry, otherwise the words would run into each other. We account for this in the
+   * indices above.
+   *
+   * @type {string}
+   * @memberof SearchIndexCache
+   */
+  readonly mergedTranscript: string = '';
+
+  readonly mergedTranscriptLowercased: string = '';
+
+  /**
+   * This gets populated as part of the search index build. It maps the start and end indicies
+   * of all of the transcript entries so we can quickly look up where an entry is in the
+   * overall transcript.
+   *
+   * @type {TranscriptEntryRange[]}
+   * @memberof SearchIndexCache
+   */
+  readonly transcriptEntryRanges: TranscriptEntryRange[] = [];
+
+  constructor(mergedTranscript = '', transcriptEntryRanges: TranscriptEntryRange[] = []) {
+    this.mergedTranscript = mergedTranscript;
+    this.transcriptEntryRanges = transcriptEntryRanges;
+    this.mergedTranscriptLowercased = mergedTranscript.toLowerCase();
+  }
+}
+
+/* eslint-disable import/prefer-default-export */
+export class SearchIndex {
+  /**
+   * Make these properties readonly. We don't want outside classes touching the private storage.
+   *
+   * @readonly
+   * @type {string}
+   * @memberof SearchIndex
+   */
+  get mergedTranscript(): string {
+    return this.searchIndexCache.mergedTranscript;
+  }
+
+  get transcriptEntryRanges(): TranscriptEntryRange[] {
+    return this.searchIndexCache.transcriptEntryRanges;
+  }
+
+  /**
+   * This gets populated as part of the search index build. It merges all of the transcript
+   * entries together so we can search it as a single document instead of a bunch of
+   * individual entries. This allows searches to cross over transcript entries.
+   *
+   * It also indexes the entries so we can quickly look them up later.
+   *
+   * @private
+   * @memberof SearchIndex
+   */
+  private searchIndexCache: SearchIndexCache = new SearchIndexCache();
```

Comment:
do we need to pass the transcript to it?

Reply:
No, in this case, it just creates an empty cache. I did this so `searchIndexCache` wasn't an optional (and having to deal with optionals all over the place). The `searchIndexCache` gets recreated with a real cache as soon as the initialization finishes so this is just a placeholder until that happens.